app/blueprints/base/views.py

```diff
@@ -62,6 +64,26 @@ def sanitize_url(url):
     return url
 
 
+def authenticated_within(max_age):
+    auth_time = session["iat"]
+    time_elapsed_since_authenticated = int(time.time()) - auth_time
+
+    return time_elapsed_since_authenticated <= max_age
+
+
+def force_authentication(path=None):
+    request_path = request.full_path
+    if path is not None:
+        request_path = path
+    session["next_url"] = request_path
+    return redirect(oidc.login("dex", force_reauthentication=True))
+
+
+@base.route('/reauthenticate/<path:caller>')
+def reauthenticate(caller=None):
+    return force_authentication(caller)
```

Comment:
This should probably be in a separate PR. Some more work is needed to make the `caller` parameter optional, because in most cases reauthentication should redirect back to the same page.

It would also be better to get the `caller` parameter from `request.args`, rather than the path (eg: `/reauthenticate?caller=/admin`), otherwise we end up with our whole URL structure being duplicated with `/reauthenticate` prefixed (eg: `/reauthenticate/details/edit` and `/reauthenticate/admin/users/1`).

Also, having a (partial) URL hardcoded in the template is not good. It would be better to either generate the URL with `url_for` or even better get it from `request.full_path` - which would then remove the need for it to be in the template at all. That might require creating a context processor function to build a reauthenticate URL.